src/org/traccar/protocol/AdmProtocolDecoder.java

```diff
@@ -82,7 +81,10 @@ protected Object decode(
             position.setAltitude(buf.readUnsignedShort());
 
             position.set(Position.KEY_HDOP, buf.readUnsignedByte() * 0.1);
-            position.set(Position.KEY_SATELLITES, buf.readUnsignedByte() & 0x0f);
+
+            int satellites = buf.readUnsignedByte() & 0x0f;
+            position.set(Position.KEY_SATELLITES, satellites);
+            position.setValid(satellites >= 3);
```

Comment:
Yes, the rule is that if device doesn't send a special validity flag, then we set it to true.